src/osx/core/uilocale.mm

```diff
@@ -130,12 +133,82 @@ int CompareStrings(const wxString& lhs, const wxString& rhs,
     return wxCFStringRef::AsString([m_nsloc localeIdentifier]);
 }
 
+wxLocaleIdent
+wxUILocaleImplCF::GetLocaleId() const
+{
+    return wxLocaleIdent::FromTag(GetName());
+}
+
 wxString
 wxUILocaleImplCF::GetInfo(wxLocaleInfo index, wxLocaleCategory cat) const
 {
     return wxGetInfoFromCFLocale((CFLocaleRef)m_nsloc, index, cat);
 }
 
+wxString
+wxUILocaleImplCF::GetLocalizedName(wxLocaleName name, wxLocaleForm form) const
+{
+    //        NSString* countryCode = [m_nsloc countryCode]
+    NSString* str = NULL;
+    switch (name)
+    {
+        case wxLOCALE_NAME_LOCALE:
+            switch (form)
+            {
+                case wxLOCALE_FORM_NATIVE:
+                    str = [m_nsloc localizedStringForLocaleIdentifier:[m_nsloc localeIdentifier]];
+                    break;
+                case wxLOCALE_FORM_ENGLISH:
+                    str = [m_nsloc displayNameForKey:NSLocaleIdentifier value:[m_nsloc localeIdentifier]];
+                    break;
+                default:
+                    wxFAIL_MSG("unknown wxLocaleForm");
+            }
+            break;
+        case wxLOCALE_NAME_LANGUAGE:
+            switch (form)
+            {
+                case wxLOCALE_FORM_NATIVE:
+                    str = [m_nsloc localizedStringForLanguageCode:[m_nsloc languageCode]];
+                    break;
+                case wxLOCALE_FORM_ENGLISH:
+                    str = [m_nsloc displayNameForKey:NSLocaleIdentifier value:[m_nsloc localeIdentifier]];
+                    break;
+                default:
+                     wxFAIL_MSG("unknown wxLocaleForm");
+            }
+            break;
+        case wxLOCALE_NAME_COUNTRY:
+            switch (form)
+            {
+                case wxLOCALE_FORM_NATIVE:
+                    str = [m_nsloc localizedStringForCountryCode:[m_nsloc countryCode]];
+                    break;
+                case wxLOCALE_FORM_ENGLISH:
+                    str = [m_nsloc displayNameForKey:NSLocaleIdentifier value:[m_nsloc localeIdentifier]];
+                    break;
+                default:
+                    wxFAIL_MSG("unknown wxLocaleForm");
+            }
+            break;
+    }
+    if (str)
+        return wxCFStringRef::AsString(str);
+    else
+        return wxString();
```

Comment:
Not really necessary, `AsString()` checks for null ref. so could simplify to just
```suggestion
    return wxCFStringRef::AsString(str);
```